sridha2412/assignment2/q2.py

```diff
@@ -2,44 +2,51 @@
 
 # node class`
 class Node:
-    # constuctor to initlalize a node object
-    def __init__(self, data):
+    def __init__(self, data: int):
+        """ constuctor to initlalize a node object """
         self.data = data
         self.left = None
         self.right = None
 
-# method to find node is present in tree
-def find_node(root, node):
+def is_present(root: Node, key: int) -> bool:
+    """ method to check if node is present in tree """
     if root is None:
         return False
 
-    if root.data == node or find_node(root.right, node) or find_node(root.left, node):
+    if root.data == key or is_present(root.right, key) or is_present(root.left, key):
         return True
 
     return False
 
-# method to find least common ancestor of two given nodes
-# note: if one key is ancestor of another, it returns the key
-def find_lca(root, node1, node2):
+def find_lca(root: Node, node1: int, node2: int) -> int:
```

Comment:
I do not understand this method, sorry, can you expand the docstring or the comments to explain the approach?

Reply:
I have added more comments to this method, hope that explains it better